### PR TITLE
Giving command buffer begin_debug_group/end_debug_group status.

### PR DESCRIPTION
**Pull request by benvanik**

### CODE REVIEW

File: runtime/src/iree/hal/command_buffer.h [hunk 1]
```diff
@@ -393,6 +393,11 @@ IREE_API_EXPORT iree_device_size_t iree_hal_collective_element_byte_count(
 // Bitfield specifying flags controlling a dispatch operation.
 enum iree_hal_dispatch_flag_bits_t {
   IREE_HAL_DISPATCH_FLAG_NONE = 0,
+
+  // Indirect parameters such as workgroup count are static at the time the
+  // command buffer is issued. This is in contrast to dynamic parameters that
+  // may be changed by dispatches within the same command buffer.
+  IREE_HAL_DISPATCH_FLAG_STATIC_INDIRECT_PARAMETERS = 1ull << 0,
```
Comment (ScottTodd) [inline runtime/src/iree/hal/command_buffer.h:400]: This new enum value looks unrelated to the other changes in this PR. Could it be moved to another PR?

Comment (benvanik) [reply runtime/src/iree/hal/command_buffer.h:400]: not worth the carbon emissions for a new enum

**Review by ScottTodd — APPROVED** [review]
